File: 0100-same-tree/0100-same-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    void helper(List<Integer> ans, TreeNode root)
    {
        if(root == null)
        {
            ans.add(Integer.MAX_VALUE);
            return;
        }
        ans.add(root.val);
        helper(ans, root.left);
        helper(ans, root.right);
    }
    public boolean isSameTree(TreeNode p, TreeNode q) {
        List<Integer> ans1 = new ArrayList<>();
        List<Integer> ans2 = new ArrayList<>();
        helper(ans1, p);
        helper(ans2, q);
        if(ans1.equals(ans2))
            return true;
        return false;
    }
}

File: 0100-same-tree/0100-same-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    // void helper(vector<int>& ans, TreeNode* root)
    // {
    //     if(root == NULL)
    //     {
    //         ans.push_back(INT_MAX);
    //         return;
    //     }
    //     ans.push_back(root->val);
    //     helper(ans, root->left);
    //     helper(ans, root->right);
    // }
    bool isSameTree(TreeNode* p, TreeNode* q) {
    //     vector<int> ans1, ans2;
    //     helper(ans1, p);
    //     helper(ans2, q);
    //     if(ans1 == ans2)
    //         return true;
    //     return false;
    // }
    if( p == NULL && q == NULL) return true;
    if( p == NULL || q == NULL) return false;
    if(p->val != q->val) return false;
    return isSameTree(p->left, q->left) && isSameTree(p->right, q->right);
    }
};